Compile fails due to `gets()`

Compile fails with `gets()` being used.

how to reproduce -
1. clone this repo
2. `cd chess_moves_of_a_knight && make KnightMain`

The mentioned error, as well as the following should show up
```
KnightMain.c:(.text+0x58): warning: the `gets' function is dangerous and should not be used.
/usr/bin/ld: KnightMain.c:(.text+0x1c): undefined reference to `createHash'
/usr/bin/ld: KnightMain.c:(.text+0xc6): undefined reference to `userinput'
```
I would love to work in this issue. please assign me :D

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,3 @@
+KnightProg
+.vscode
+*.o

--- a/KnightMain.c
+++ b/KnightMain.c
@@ -8,13 +8,16 @@
 int main()
 {
 	Hash *table=createHash();
+
 	printf("The position in a chess board is given as A1,A2,A3...B1,B2,B3...H6,H7,H8 where the alphabet represents the row number and the number represents the column number.\nThe values returned on entering the position when prompted is the next possible positions that a knight can move on a chess board.\n");
+
 	while(1)
 	{
 		char x[10];
 		printf("\n");
 		printf("Enter position on chess board. Enter 0 to exit:");
-		gets(x);
+		// gets(x);
+		scanf("%s", x);
 		if(*x=='0')
 			break;
 		else if(strlen(x)!=2)
@@ -26,6 +29,8 @@ int main()
 		userinput(x[0],y,table);
 		fflush(stdin);
 	}
+
 	free(table);
+
 	return 0;
 }

--- a/KnightMoves.h
+++ b/KnightMoves.h
@@ -6,7 +6,7 @@ typedef struct Position
 	int num_pos;		//Position given by the user converted into integer form
 	int num_of_knight_next; //Number of next knight positions
 	int knight_next[8]; //Integer array consisting of next knight moves
-}Position;
+} Position;
 
 /*Struct to define the hash table*/
 typedef struct Hash

--- a/README.md
+++ b/README.md
@@ -4,3 +4,13 @@ Given the position of a knight list down all possible next moves.
 Although chess is a very interesting two player game a chess board is also equally interesting as it opens doors to various computational problems. The moves of the knight are rather unusual and make a great programming problem.
 # Data Structure
 Two-dimensional matrix
+
+# Compiling
+```
+make -f KnightMake
+```
+
+# Usage
+```
+./KnightProg
+```

